src/dist/common_providers/partition_resolver_simple.cpp

```diff
@@ -94,9 +94,9 @@ namespace dsn
         void partition_resolver_simple::on_access_failure(int partition_index, error_code err)
         {
             if (-1 != partition_index
-                && err != ERR_CAPACITY_EXCEEDED
-                && err != ERR_NOT_ENOUGH_MEMBER
-                && err != ERR_INVALID_DATA)
+                && err != ERR_CAPACITY_EXCEEDED // no need for reconfiguration on primary
+                && err != ERR_NOT_ENOUGH_MEMBER // primary won't change and we only r/w on primary in this provider
```

Comment:
If there is an INVALID_DATA error, the local replica should be turned into PS_ERROR state, therefore we have to clear the partition cache and get the new ones.